Separate past volunteers on team page

fbalda opened the pull request:
Make the team page fetch the team data from our new endpoints instead of a json file and visually separate current and past volunteers.
Also adds dummy data for testing with local backend, and some code related to updating the user's volunteer info (though the feature is not in this PR).

> [!WARNING]  
> Should only be merged after we have up-to-date team data in our backend

Closes #5946

**Web frontend checklist**
- [x] Formatted my code with `yarn format`
- [x] There are no warnings from `yarn lint --fix`
- [x] There are no console warnings when running the app
- [x] Added tests where relevant
- [x] All tests pass
- [x] Clicked around my changes running locally and it works
- [x] Checked Desktop, Mobile and Tablet screen sizes

**Other**
*Untick the following if you'd prefer that maintainers don't push commits/merge your branch.*
- [x] A maintainer can push commits to my branch
- [x] A maintainer can merge my PR (you can also merge after approval)

fbalda committed:
Updated team page with past contributor section, added show volunteer profile setting

fbalda committed:
Merge branch 'develop' into web/feature/new-team-page

fbalda committed:
Remove superfluous script

fbalda committed:
Cleanup

fbalda committed:
Formatting

aapeliv commented:
It's in the db now @fbalda

fbalda committed:
Merge branch 'develop' into web/feature/new-team-page

fbalda commented:
> It's in the db now @fbalda

Is it only in prod? It's empty on staging for me. I think we should have the data on staging, too, so we can test properly before pushing to prod

nabramow commented:
> > It's in the db now @fbalda
> 
> Is it only in prod? It's empty on staging for me. I think we should have the data on staging, too, so we can test properly before pushing to prod

Hey @fbalda I just added some test data to the volunteers table on staging. Can't copy directly from prod as the userids and usernames are different. I think I added some for every state now: not visible on team page, visible, current volunteer, past volunteer, with website and without.

fbalda committed:
Styling update

fbalda committed:
Merge branch 'develop' into web/feature/new-team-page

fbalda committed:
React query fix

fbalda committed:
Cleanup

fbalda commented:
@nabramow Ready for review

Edit: Still need to fix the backend formatting
Edit Edit: Done

fbalda committed:
Fix backend styling

fbalda committed:
Merge branch 'develop' into web/feature/new-team-page

nabramow commented:
Just some small design stuff. Can we make the volunteer cards align left, make the cards always the same size even if they don't have a link, and maybe make the section headings a bit bigger?

<img width="1262" height="576" alt="Screenshot 2025-09-05 at 8 12 22 AM" src="https://github.com/user-attachments/assets/3db285ce-2d2e-479f-ba99-77d97498289e" />

<img width="1210" height="303" alt="Screenshot 2025-09-05 at 8 12 32 AM" src="https://github.com/user-attachments/assets/f1a19151-5426-4906-a343-e84ad92f88da" />

fbalda committed:
Team page styling update

fbalda commented:
> Just some small design stuff. Can we make the volunteer cards align left, make the cards always the same size even if they don't have a link, and maybe make the section headings a bit bigger?
> <img alt="Screenshot 2025-09-05 at 8 12 22 AM" width="1262" height="576" src="https://private-user-images.githubusercontent.com/18261566/486165605-3db285ce-2d2e-479f-ba99-77d97498289e.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTcyMzMxMTksIm5iZiI6MTc1NzIzMjgxOSwicGF0aCI6Ii8xODI2MTU2Ni80ODYxNjU2MDUtM2RiMjg1Y2UtMmQyZS00NzlmLWJhOTktNzdkOTc0OTgyODllLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA5MDclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwOTA3VDA4MTMzOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWMxNjc2ZGNjN2I4MmYyOTQ3MGI3NGQ2MTYzYWJkODgxZTUyMzZlZDAxN2I4NTNlMjc1ZWVkZjNhYTAwN2FiOWQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.eSyeFiqH-JPUtR0blsUKzTFlVXhrhu_BtV4KdGLnCqo"> <img alt="Screenshot 2025-09-05 at 8 12 32 AM" width="1210" height="303" src="https://private-user-images.githubusercontent.com/18261566/486165644-f1a19151-5426-4906-a343-e84ad92f88da.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTcyMzMxMTksIm5iZiI6MTc1NzIzMjgxOSwicGF0aCI6Ii8xODI2MTU2Ni80ODYxNjU2NDQtZjFhMTkxNTEtNTQyNi00OTA2LWEzNDMtZTg0YWQ5MmY4OGRhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA5MDclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwOTA3VDA4MTMzOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTZhMzM1Y2FmM2E4NzM0Njg4MDYxZTJiZDBiNjMxMDgyZjQ2OTAxZTYxYmIzYjgyYmUyMjVhNGU2OWExZjE3MTImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.vcMENRK_KmaWuYKsmA5ytVGZVYRu3Zay1DRfnF9xXQY">

Yeah, I originally just took the styling as it was and didn't change anything. Updated it now to include your suggestions but also made lots of other tweaks and changes to try fit our current flat design language better and improve readability. (Though I took a min-max approach as we probably want to do a complete redesign of the page and unify the styling of components across the app in the not-so-distant future).

~Note that I changed the layout to list board members first, then leads, then the rest, with a bit of spacing between them. That's why there is now a larger gap between the rows when rendering the current staging db data. Maybe you can add a few more team members, so we can see how it looks when there are multiple rows in a section? That would be awesome 🙏~

Edit: I tested with prod data and the styling for board members was a bit overwhelming when there's 3 rows of them, so I dialed it back. Also I only separate board members into their own section now, as the lead section as implemented before only contained Chris with prod data. Maybe we can discuss if we want some additional visual hierarchy in the future

fbalda committed:
Team page styling update

nabramow commented:
Okay mostly style stuff again!

- I think we should give headings "Board Members", "Current volunteers", "Past volunteers" as for the first two sections it's not so clear why it's separated.
- Let's set a maxWidth to the cards maybe? Lots of extra space to the right:

<img width="1200" height="600" alt="Screenshot 2025-09-07 at 8 37 40 AM" src="https://github.com/user-attachments/assets/48ec7493-ad8c-4840-a89e-928a42846db3" />

- Maybe also a height so the cards are always the same height even if no website given

- What do you think about adding some box shadow? Can use theme.shadow[4] think it what we normally use.

- When the headings are indented to the right of the end of the cards, it looks a little strange on the laptop screen:

<img width="1203" height="440" alt="Screenshot 2025-09-07 at 8 40 04 AM" src="https://github.com/user-attachments/assets/f9228794-d556-4eca-8f2c-aec971455236" />

fbalda commented:
@nabramow 
Ok, so going through your points 😄 

> * I think we should give headings "Board Members", "Current volunteers", "Past volunteers" as for the first two sections it's not so clear why it's separated.

and
> * Maybe also a height so the cards are always the same height even if no website given

So until now we had the extra line "Board Member" on the board member cards, also board members had their name written in a larger font. This looked a bit cluttered and readability suffered, so I changed the "Board Member" text to a badge and reduced the size of board member names. However when setting a fixed height for cards, this created a lot of white space on the bottom of non-board member cards without a link, so I changed the the size for board member cards to be a bit larger. All board member cards have the same height and all non-board member cards have the same height.

IMO it doesn't look too confusing, as core people are usually listed first and they all have the "Board Member" badge: First comes the team, with board members at the top and then below we credit former contributors. But I don't have a strong opinion here. However, if we add headings, I would remove the redundant "Board Member" badges.

> * Let's set a maxWidth to the cards maybe? Lots of extra space to the right:

Agreed :+1:

> * What do you think about adding some box shadow? Can use theme.shadow[4] think it what we normally use.

So I intentionally removed shadows, as they IMO don't fit with the flat design direction we are moving to with the new page redesigns (at least with the values defined in our theme, I think a more subtle shadow could work better). 
I think we really should push for a CI/design document, laying out guidelines and the look we are going for. It feels wrong to do stuff like this on a page-by-page or even component-by-component basis. I can add it back, but IMO `theme.shadow[4]` looks very intense, I'd probably use `theme.shadow[1]`.

> * When the headings are indented to the right of the end of the cards, it looks a little strange on the laptop screen:

I agree, but it also looks strange to have it not indented, while the top and bottom paragraphs are. Not sure why it was done this way, so I kind of didn't want to touch it :smile:, but I'll change the top and bottom paragraphs to be wider and make everything aligned.

I'd really like to streamline the design process, might be worth to invest some time now to save a lot later. Let's discuss in the next dev meeting, maybe I can make a concrete proposal (no promises yet 😄).

nabramow commented:
> @nabramow Ok, so going through your points 😄

> So until now we had the extra line "Board Member" on the board member cards, also board members had their name written in a larger font. This looked a bit cluttered and readability suffered, so I changed the "Board Member" text to a badge and reduced the size of board member names. However when setting a fixed height for cards, this created a lot of white space on the bottom of non-board member cards without a link, so I changed the the size for board member cards to be a bit larger. All board member cards have the same height and all non-board member cards have the same height.
> 
> IMO it doesn't look too confusing, as core people are usually listed first and they all have the "Board Member" badge: First comes the team, with board members at the top and then below we credit former contributors. But I don't have a strong opinion here. However, if we add headings, I would remove the redundant "Board Member" badges.

What if we just added "Current Volunteers" at the top then? Since we have past volunteers. Either that our "Our Team". Then center the headings to solve the indentation issue?

> I agree, but it also looks strange to have it not indented, while the top and bottom paragraphs are. Not sure why it was done this way, so I kind of didn't want to touch it 😄, but I'll change the top and bottom paragraphs to be wider and make everything aligned.
> 
> I'd really like to streamline the design process, might be worth to invest some time now to save a lot later. Let's discuss in the next dev meeting, maybe I can make a concrete proposal (no promises yet 😄).

Yeah for sure we def have room for improvement here. Life is rough without a designer ahah.

nabramow commented:
Hey @fbalda - since you're busy refactoring would you be okay if I push the last style changes to this PR? Tryna cross the fall deadline PRs off the list and this one is so close and I was the one complaining about style anyway ahah.

fbalda commented:
> Hey @fbalda - since you're busy refactoring would you be okay if I push the last style changes to this PR? Tryna cross the fall deadline PRs off the list and this one is so close and I was the one complaining about style anyway ahah.

Please do, happy to get this done 🙏

nabramow committed:
Fix merge conflicts

nabramow committed:
Add missing bracket

nabramow committed:
Make style adjustments to team page

nabramow reviewed (approved):
I made some small style adjustments as discussed! Good to go I think!
